app/deepseg.cc

```diff
@@ -410,18 +470,24 @@ int main(int argc, char* argv[]) try {
 	printf("flip_h: %s\n", flipHorizontal ? "yes" : "no");
 	printf("flip_v: %s\n", flipVertical ? "yes" : "no");
 	printf("threads:%zu\n", threads);
-	printf("back:   %s\n", back ? back : "(none)");
-	printf("model:  %s\n\n", modelname);
+	printf("back:   %s => %s\n", back ? back : "(none)", s_backg.empty() ? "(none)" : s_backg.c_str());
+	printf("model:  %s => %s\n\n", modelname ? modelname : "(none)", s_model.empty() ? "(none)" : s_model.c_str());
+
+	// No model - stop here
+	if (s_model.empty()) {
+		printf("Error: unable to load specified model: %s\n", modelname);
+		exit(1);
+	}
 
 	// Create debug window early (ensures highgui is correctly initialised on this thread)
 	if (debug > 1) {
 		cv::namedWindow(DEBUG_WIN_NAME, cv::WINDOW_AUTOSIZE | cv::WINDOW_GUI_EXPANDED);
 	}
 
 	// Load background if specified
-	auto pbk((back) ? load_background(back, debug) : nullptr);
+	auto pbk((s_backg.empty()) ? nullptr : load_background(s_backg.c_str(), debug));
```

Comment:
```suggestion
	auto pbk(s_backg.empty() ? nullptr : load_background(s_backg.c_str(), debug));
```

Reply:
along with `std::optional` return type change above, then `s_backg` and `s_model` are also optional so this should clean up to avoid `.empty()` checks...